### PR TITLE
Update Postgres versions

### PR DESCRIPTION
**Pull request by tbalthazar**
- Update postgres to the latest patch versions

### CODE REVIEW

**Comment by tbalthazar** [comment]
```
$ aws s3 ls s3://ci-database-binary/ --recursive | grep 2020-02-14
2020-02-14 14:01:07         32 .md5/cedar-14/postgresql-10.12.tgz.md5
2020-02-14 14:33:32         32 .md5/cedar-14/postgresql-11.7.tgz.md5
2020-02-14 15:06:38         32 .md5/cedar-14/postgresql-12.2.tgz.md5
2020-02-14 13:34:27         32 .md5/cedar-14/postgresql-9.6.17.tgz.md5
2020-02-14 14:15:28         32 .md5/heroku-16-build/postgresql-10.12.tgz.md5
2020-02-14 14:46:57         32 .md5/heroku-16-build/postgresql-11.7.tgz.md5
2020-02-14 15:20:50         32 .md5/heroku-16-build/postgresql-12.2.tgz.md5
2020-02-14 13:44:15         32 .md5/heroku-16-build/postgresql-9.6.17.tgz.md5
2020-02-14 14:09:51         32 .md5/heroku-16/postgresql-10.12.tgz.md5
2020-02-14 14:40:55         32 .md5/heroku-16/postgresql-11.7.tgz.md5
2020-02-14 15:15:03         32 .md5/heroku-16/postgresql-12.2.tgz.md5
2020-02-14 13:39:19         32 .md5/heroku-16/postgresql-9.6.17.tgz.md5
2020-02-14 14:28:04         32 .md5/heroku-18-build/postgresql-10.12.tgz.md5
2020-02-14 15:00:54         32 .md5/heroku-18-build/postgresql-11.7.tgz.md5
2020-02-14 15:33:51         32 .md5/heroku-18-build/postgresql-12.2.tgz.md5
2020-02-14 13:55:47         32 .md5/heroku-18-build/postgresql-9.6.17.tgz.md5
2020-02-14 14:22:15         32 .md5/heroku-18/postgresql-10.12.tgz.md5
2020-02-14 14:54:07         32 .md5/heroku-18/postgresql-11.7.tgz.md5
2020-02-14 15:27:17         32 .md5/heroku-18/postgresql-12.2.tgz.md5
2020-02-14 13:49:44         32 .md5/heroku-18/postgresql-9.6.17.tgz.md5
2020-02-14 14:00:12   24168383 cedar-14/postgresql-10.12.tgz
2020-02-14 14:32:38   25095341 cedar-14/postgresql-11.7.tgz
2020-02-14 15:05:39   27404799 cedar-14/postgresql-12.2.tgz
2020-02-14 13:33:13   22605413 cedar-14/postgresql-9.6.17.tgz
2020-02-14 14:14:32   24620461 heroku-16-build/postgresql-10.12.tgz
2020-02-14 14:45:32   25612603 heroku-16-build/postgresql-11.7.tgz
2020-02-14 15:19:51   27923930 heroku-16-build/postgresql-12.2.tgz
2020-02-14 13:43:25   23047993 heroku-16-build/postgresql-9.6.17.tgz
2020-02-14 14:05:38   24620929 heroku-16/postgresql-10.12.tgz
2020-02-14 14:38:08   25612824 heroku-16/postgresql-11.7.tgz
2020-02-14 15:11:20   27923451 heroku-16/postgresql-12.2.tgz
2020-02-14 13:38:31   23048413 heroku-16/postgresql-9.6.17.tgz
2020-02-14 14:27:09   26874544 heroku-18-build/postgresql-10.12.tgz
2020-02-14 14:59:12   27822257 heroku-18-build/postgresql-11.7.tgz
2020-02-14 15:32:37   30412601 heroku-18-build/postgresql-12.2.tgz
2020-02-14 13:54:19   25023337 heroku-18-build/postgresql-9.6.17.tgz
2020-02-14 14:20:38   26874463 heroku-18/postgresql-10.12.tgz
2020-02-14 14:52:10   27821205 heroku-18/postgresql-11.7.tgz
2020-02-14 15:26:10   30413729 heroku-18/postgresql-12.2.tgz
2020-02-14 13:48:53   25023084 heroku-18/postgresql-9.6.17.tgz
```

**Comment by tbalthazar** [comment]
This update also addresses CVE-2020-1720:
https://github.com/heroku/platsec/issues/1515#issuecomment-587686221

**Comment by tbalthazar** [comment]
As of today, default versions of PG addon haven't been updated to the latest version:
```console
$ h addons:create heroku-postgresql -a test-pg-versions
Creating heroku-postgresql on ⬢ test-pg-versions... free
Database has been created and is available
 ! This database is empty. If upgrading, you can transfer
 ! data from another database with pg:copy
Created postgresql-encircled-94081 as DATABASE_URL
Use heroku addons:docs heroku-postgresql to view documentation
$ h pg:info -a test-pg-versions
=== DATABASE_URL
Plan:                  Hobby-dev
Status:                Available
Connections:           0/20
PG Version:            12.1
Created:               2020-03-03 08:27 UTC
Data Size:             7.9 MB
Tables:                0
Rows:                  0/10000 (In compliance)
Fork/Follow:           Unsupported
Rollback:              Unsupported
Continuous Protection: Off
Add-on:                postgresql-encircled-94081

$ h addons:create heroku-postgresql --version=11 -a test-pg-versions
Creating heroku-postgresql on ⬢ test-pg-versions... free
Database has been created and is available
 ! This database is empty. If upgrading, you can transfer
 ! data from another database with pg:copy
Created postgresql-defined-09074 as DATABASE_URL
Use heroku addons:docs heroku-postgresql to view documentation
$ h pg:info -a test-pg-versions
=== DATABASE_URL
Plan:                  Hobby-dev
Status:                Available
Connections:           0/20
PG Version:            11.6
Created:               2020-03-03 08:29 UTC
Data Size:             7.7 MB
Tables:                0
Rows:                  0/10000 (In compliance)
Fork/Follow:           Unsupported
Rollback:              Unsupported
Continuous Protection: Off
Add-on:                postgresql-defined-09074
```

**Comment by tbalthazar** [comment]
I confirmed with @beanieboi that only the hobby plans still are on 12.1.
Standard plans and above have already been updated.

File: support/pg_versions [hunk 1]
```diff
@@ -1 +1 @@
-12.1
+9.6.17;10.12;11.7;12.2
```
Comment (bensymonds) [inline support/pg_versions]: I don't really understand this change. Why is not:

```diff
- 9.6.12;10.7;11.2;12.1
+ 9.6.17;10.12;11.7;12.2
```

Comment (tbalthazar) [reply support/pg_versions]: This `pg_versions` file was introduced in #32 where a small refactoring of the build script was done, and pg 12.1 was built to set it as the new default version.

Comment (bensymonds) [reply support/pg_versions]: Ah ok, sorry, I missed the previous PR. So this file is just used at build time and contains the pg versions we last built. In #32 we only needed 12.1 but now we need all the rest. I see.

Comment (tbalthazar) [reply support/pg_versions]: > Ah ok, sorry, I missed the previous PR. 

NP, I should have explicitly linked to #32 for clarity.

> So this file is just used at build time and contains the pg versions we last built. In #32 we only needed 12.1 but now we need all the rest. I see.

Exactly that.

**Pull request merged**